bids/variables/variables.py

```diff
@@ -447,18 +447,33 @@ def resample(self, sampling_rate, inplace=False, kind='linear'):
 
         self.index = self._build_entity_index(self.run_info, sampling_rate)
 
-        x = np.arange(n)
         num = len(self.index)
 
-        from scipy.interpolate import interp1d
-        f = interp1d(x, self.values.values.ravel(), kind=kind)
-        x_new = np.linspace(0, n - 1, num=num)
-        self.values = pd.DataFrame(f(x_new))
+        if integration_window is not None:
+            # Number of timesteps in existing time series
+            n_frames = int(np.floor(integration_window * old_sr))
+            old_times = np.arange(n) / old_sr
+            new_times = np.arange(num) / sampling_rate
+            diffs = old_times.reshape((1, -1)) - new_times.reshape((-1, 1))
+            integrator = ((diffs >= 0) & (diffs < integration_window)).astype(np.uint8)
+            del diffs
+
+            old_vals = self.values.values
+            self.values = pd.DataFrame(
+                integrator.dot(old_vals) / integrator.sum(1, keepdims=True))
+        else:
+            from scipy.interpolate import interp1d
+            x = np.arange(n)
+            f = interp1d(x, self.values.values.ravel(), kind=kind)
```

Comment:
this is ok for upsampling not fine for downsampling. depending on frequency content this will introduce aliasing.

Reply:
I'm not sure avoiding aliasing is in scope... or at least, I'm not sure we want to implicitly start low-pass filtering the input signal without the user's explicit consent. I'd be okay adding an explicit `TemporalFilter` transformation (actually I think this is already in the spec and just not yet implemented), and the user can then call that themselves before the resampling step if they like. But doing some magic internally to pick a suitable filter or otherwise avoid aliasing means (a) reproducibility across packages is limited, and (b) we're taking on the responsibility for producing sensible results, and to my mind this really falls on the user.